src/Microsoft.OpenApi.Readers/V3/OpenApiV3VersionService.cs

```diff
@@ -71,6 +71,7 @@ public OpenApiV3VersionService(OpenApiDiagnostic diagnostic)
             string reference,
             ReferenceType? type)
         {
+            var openApiReference = new OpenApiReference();
```

Comment:
you're not using this variable at all in the if block, might be worth moving it to the else to avoid unecessary allocations?